src/OpenLoco/Paint/PaintMiscEntity.cpp

```diff
@@ -0,0 +1,208 @@
+        {
+            session.addToPlotListAsParent(imageId, { 0, 0, base->z }, { 1, 1, 0 });
+        }
+        else
+        {
+            session.addToPlotListAsParent(imageId, { 0, 0, base->z }, { 1, 1, 0 }, { -12, -12, base->z });
+        }
+    }
+
+    static void paintRedGreenCurrencyEntity(PaintSession& session, MiscBase* base)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        int8_t* wiggleYOffsets = reinterpret_cast<int8_t*>(0x4FAAC8);
+        if (dpi->zoom_level > 1)
+        {
+            return;
+        }
+        MoneyEffect* moneyEffect = base->asRedGreenCurrency();
+        const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_green : StringIds::format_currency_expense_red_negative;
+        uint32_t currencyAmount = abs(moneyEffect->amount);
+        int8_t* yOffsets = &wiggleYOffsets[moneyEffect->wiggle];
+
+        session.addToStringPlotList(currencyAmount, stringId, moneyEffect->y, moneyEffect->z, yOffsets, moneyEffect->offsetX);
+    }
+
+    static void paintWindowCurrencyEntity(PaintSession& session, MiscBase* base)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        int8_t* wiggleYOffsets = reinterpret_cast<int8_t*>(0x4FAAC8);
+        if (dpi->zoom_level > 1)
+        {
+            return;
+        }
+        MoneyEffect* moneyEffect = base->asWindowCurrency();
+        const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_in_company_colour : StringIds::format_currency_expense_in_company_colour_negative;
+        uint32_t currencyAmount = abs(moneyEffect->amount);
+        int8_t* yOffsets = &wiggleYOffsets[moneyEffect->wiggle];
+        uint16_t companyColour = CompanyManager::getCompanyColour(moneyEffect->var_2E);
+
+        session.addToStringPlotList(currencyAmount, stringId, moneyEffect->y, moneyEffect->z, yOffsets, moneyEffect->offsetX, companyColour);
+    }
+
+    static void paintVehicleCrashParticleEntity(PaintSession& session, MiscBase* base)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        if (dpi->zoom_level != 0)
+        {
+            return;
+        }
+
+        VehicleCrashParticle* particle = base->asVehicleCrashParticle();
+
+        loco_global<int32_t[5], 0x4FAAB4> vehicle_particle_base_sprites;
+        uint32_t imageId = vehicle_particle_base_sprites[particle->crashedSpriteBase] + particle->frame / 256;
+        imageId = Gfx::recolour2(imageId, particle->colourScheme);
+
+        session.addToPlotListAsParent(imageId, { 0, 0, particle->z }, { 1, 1, 0 });
+    }
+
+    static void paintExplosionCloudEntity(PaintSession& session, MiscBase* base)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        if (dpi->zoom_level > 2)
+        {
+            return;
+        }
+        ExplosionCloud* particle = base->asExplosionCloud();
+        uint32_t imageId = 3372 + (particle->frame / 256);
```

Comment:
We'd like to avoid hardcoding and adding image ids, as we'll eventually be switching to a different format. Could you add constants to `Graphics/ImageIds.h` and a lookup table for them? An example of this elsewhere in the codebase would be:

https://github.com/OpenLoco/OpenLoco/blob/e8fdae378631c4c4aff3bdf6d2e84174b1f450cb/src/OpenLoco/Windows/CompanyList.cpp#L1537-L1552

Reply:
So would you like to define 256 frames (constants) in this case?

Reply:
Heh, that was my initial thought as well, but 256 is just the step size. It's not 256 frames, more like 18 in this case.

It may help to make a dump of the g1 images to easily inspect them.